File: NERP.Admin.Web1/ClientApp/src/app/entities/models/odf.ts
import { Point } from '../../common/geo/geometry';
import { BaseNetworkEntity } from './baseNetworkEntity';
import { Installation } from './installation';
import { Owner } from './owner';

export class ODF extends BaseNetworkEntity {
    noOfPorts?: number;
    mountingType?: string;
    height?: number;
    width?: number;
    length?: number;

    geometry: Point;
    installation?: Installation = new Installation();
    owner?: Owner = new Owner();
}

File: NERP.NetworkEntities/Models/ODF.cs
using NERP.Common.Models;

namespace NERP.NetworkEntities.Models
{
    public class ODF : BaseNetworkEntity
    {
        public Point Geometry { get; set; }
        public int? NoOfPorts { get; set; }
        public string MountingType { get; set; }
        public double Height { get; set; }
        public double Weidth { get; set; }
        public double Length { get; set; }
        public Installation Installation { get; set; }
        public Owner Owner { get; set; }

    }
}
